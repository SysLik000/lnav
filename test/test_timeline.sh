#! /bin/bash

export YES_COLOR=1

export TZ=UTC

touch -t 200711030923 ${srcdir}/logfile_glog.0
run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ":hide-fields log_opid" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ";SELECT * FROM all_opids" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-in CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-out CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-file *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':close *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-before 2011-11-03 00:19:37' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-after 2011-11-03 00:20:30' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_strace_log.1
