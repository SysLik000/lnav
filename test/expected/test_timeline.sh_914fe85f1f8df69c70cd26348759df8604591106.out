[1m[4m[35m   Duration   [0m[4m|[0m[4m [0m[1m[4m[31mâœ˜[0m[4m[33mâ–²[0m[4m [0m[4m|[0m[1m[4m[35m Operation[0m[4m  [0m[4m|[0m[4m052      [0m[4m|[0m[4m104      [0m[4m|[0m[4m156      [0m[4m|[0m[4m208[0m
[32m         421us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m5d[0m[7m[32mf[0m[32m3ebceffba6647931c43adc9b66174[0m[32m execve[0m
[32m           9us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m0e[0m[7m[32m6[0m[32md57ceebc0639747f65f587090c2a3[0m[32m brk[0m
[1m[32m           3us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m29[0m[1m[7m[32mb[0m[1m[32m749ca655def7af91220f2bb15147d[0m[1m[32m arch_prctl[0m
[1m[32m           2us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m49[0m[1m[7m[32m7[0m[1m[32m1a5b88d403b63278c7afe83a4fe41[0m[1m[32m set_tid_address[0m
[32m          15us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mf7[0m[7m[32m6[0m[32m2ce9f997ffff2f9b6002f8838a8d8[0m[32m socket[0m
[32m           4us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m50[0m[7m[32md[0m[32mc7bba20b5694f5394bdd950005c4b[0m[32m setsockopt[0m
[1m[32m          42us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m1c[0m[1m[7m[32m9[0m[1m[32m1be72d460459cb4c3640d3a1fb2a3[0m[1m[32m bind[0m
[1m[32m           8us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m76[0m[1m[7m[32m3[0m[1m[32m5ae3f4a4aa36e6f17f3f951f5bef6[0m[1m[32m listen[0m
[32m          55us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m42[0m[7m[32m8[0m[32me080fa3f5277bdaf8d65ed67341c4[0m[32m clone[0m
[32m           4us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32ma3[0m[7m[32m9[0m[32ma3cf77138c513da0a96dc8483393e[0m[32m set_robust_list[0m
[1m[32m         513us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m74[0m[1m[7m[32m9[0m[1m[32mb8ad73499897e3e7c83af15ed0660[0m[1m[32m futex[0m
[1m[32m           002[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m0f[0m[1m[7m[32mb[0m[1m[32m1f64bcca16e1a9434cb4d3337c816[0m[1m[32m accept4[0m
[32m          86us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32md7[0m[7m[32mb[0m[32mf91c1e0c85e657c686bf34c83f903[0m[32m clone[0m
[32m           5us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m37[0m[7m[32m4[0m[32mcef0f21af5522a3222da79e388ad1[0m[32m set_robust_list[0m
[1m[32m         212us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m63[0m[1m[7m[32me[0m[1m[32m00f4367e2d68b27680977724bed5c[0m[1m[32m read[0m
[1m[32m          44us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m05[0m[1m[7m[32mc[0m[1m[32m58ef7ad905c41335e763ce0a85159[0m[1m[32m openat[0m
[32m          12us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m40[0m[7m[32ma[0m[32mbd632368715b4f1f585a81b69192b[0m[32m read[0m
[32m          53us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m26[0m[7m[32mb[0m[32m9817833fff0719b2f5fe6cbd5c62f[0m[32m write[0m
[1m[32m           7us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32ma5[0m[1m[7m[32m9[0m[1m[32mf0c6f0618876ca67d26d019047a12[0m[1m[32m close[0m
[1m[32m           6us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m0b[0m[1m[7m[32mc[0m[1m[32mf443cac828f6b09da5ccef4533260[0m[1m[32m close[0m
[32m         177us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m69[0m[7m[32md[0m[32m5d9dee9ec09559b0ebffea3d935f7[0m[32m wait4[0m
[32m          21us[0m[32m  [0m[1m[31mâ–ˆ[0m[33m [0m[32m  [0m[7m[32m4[0m[32m3e8ac1e8e627b7119b4b4d9e7840a27[0m[32m futex[0m
[1m[32m           050[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mf5[0m[1m[7m[32m666e24cc3d[0m[1m[32m49ab63434e2f18f77042[0m[1m[32m clock_nanosleep[0m
[1m[32m           006[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32ma6df[0m[1m[7m[32m0[0m[1m[32m4ebbee9964bd61ea76da4ea957a[0m[1m[32m accept4[0m
[32m          83us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m86e3[0m[7m[32m7[0m[32m3ae97fd9f468256f95dcc4788d2[0m[32m clone[0m
[32m           4us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m2741[0m[7m[32m1[0m[32m677970b8d3882a3c4d72c42df24[0m[32m set_robust_list[0m
[1m[32m         186us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mab33[0m[1m[7m[32m5[0m[1m[32m6b2c156e09f33941d285b885ac1[0m[1m[32m read[0m
[1m[32m          17us[0m[1m[32m  [0m[1m[31mâ–ˆ[0m[1m[33m [0m[1m[32m  [0m[1m[32m9e[0m[1m[7m[32mc[0m[1m[32mc4464c119182ec80d8f2b10024001[0m[1m[32m openat[0m
[32m          45us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m5886[0m[7m[32mc[0m[32m25878b3a58cdebe037561622e24[0m[32m write[0m
[32m           7us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m38b0[0m[7m[32m7[0m[32m26e748a87fccf9c108d32a12bac[0m[32m close[0m
[1m[32m         144us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mc2c9[0m[1m[7m[32m4[0m[1m[32me26227a2631c87ff07420b3ea94[0m[1m[32m wait4[0m
[1m[32m           4us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m62ee[0m[1m[7m[32m1[0m[1m[32mdd6ec4adf9417910faafb3d80cd[0m[1m[32m gettimeofday[0m
[32m           6us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m1984[0m[7m[32md[0m[32me5445e87b60d5ca82b2f6062e75[0m[32m futex[0m
[32m          34us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m5a1f[0m[7m[32m8[0m[32m7fe43362d98a90438f28f968908[0m[32m futex[0m
[1m[32m           9us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m0563[0m[1m[7m[32md[0m[1m[32mfc6d6b5e702b6cd88106ab784d4[0m[1m[32m write[0m
[1m[32m          23us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m4fa1[0m[1m[7m[32mc[0m[1m[32m58885b4fce1046d478769f74b05[0m[1m[32m openat[0m
[32m          11us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m1d09[0m[7m[32m8[0m[32m0b1bb59fa62d5fe4673b70d5e1d[0m[32m write[0m
[32m           001[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m88b2[0m[7m[32m4[0m[32mebf4b399da3702d7e5bcc01b794[0m[32m fsync[0m
[1m[32m           7us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m2808[0m[1m[7m[32m6[0m[1m[32m6aff8f9fc2b208236d57ebd9660[0m[1m[32m close[0m
[1m[32m           100[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m2563[0m[1m[7m[32md51267aa04d39df2561[0m[1m[32ma7e7f974c[0m[1m[32m nanosleep[0m
[32m           102[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mff96206cc32e28bd6bec2f9[0m[7m[32mf5e04c6b3[0m[7m[32m accept4[0m
[32m          91us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m3b152a7eb716b8487489e32727248dc5[0m[32m clone[0m
[1m[32m           4us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m3cf993b6737e61a403240bba03263889[0m[1m[32m set_robust[0m[1m[7m[32m_[0m[1m[32mlist[0m
[1m[32m         178us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m7a6209795ad3df264c2b37e43cfc6510[0m[1m[32m read[0m
[32m          37us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m3e19c4e47a4f0f1f20dcce7196b7977a[0m[32m openat[0m
[32m          10us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mcecc937e077ff75cfe25e18634e98d0f[0m[32m write[0m
[1m[32m           6us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m50fdb8bcb45a54626995588967fb6264[0m[1m[32m close[0m
[1m[32m          18us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m6553007c0e7485e37ee3aa91eb22833b[0m[1m[32m write[0m
[32m           6us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mbbd538bc54b73abdd756847be6484dcd[0m[32m close[0m
[32m         182us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m3b166a24a7610bb1b44c820d9c1e1036[0m[32m wait4[0m
[1m[32m           5us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m01c4900ffa973404bad3f20bca06bec7[0m[1m[32m futex[0m
[1m[32m          29us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m0d94723d4c8a944a79af94ad6177a8bd[0m[1m[32m futex[0m
[32m           9us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m3db1d79d03e535d80951d7981a2ea404[0m[32m write[0m
[32m           7us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mb3d63e13b1ce00c7bd9beb34073ef916[0m[32m close[0m
[1m[32m           6us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32m968b03085f35aec18c4b44f7a1b00bda[0m[1m[32m close[0m
